Automattic/wp-calypso
# SitesOverview: split utils functions into separate files

This pull request extracts all the functions and constants inside `client/jetpack-cloud/sections/agency-dashboard/sites-overview/utils.ts` into separate files, and organizes them so that they're located nearer to where they're used.

The code itself is unchanged by-and-large, but some refactoring was necessary to mitigate the use of the raw `translate` function and replace it with `useTranslate`, as recommended in the `i18n-calypso` package documentation.

Resolves https://github.com/Automattic/jetpack-genesis/issues/16.

## Proposed Changes

* Extract functions and constants from `utils.ts` into various new files. Generally, `lib` directories are for non-hook helper functions and `hooks` are for React hooks.
* Replace all references to `utils.ts` with updated references to the newly-relocated code.
* Migrate and update all unit test cases for existing functions.
* Remove `utils.ts`.

## Testing Instructions

* Examine the code changes in this PR commit-by-commit, verifying that the changes at each step make sense.
* Ensure all unit and integration tests pass.
* Ensure all type checks pass.
* Fully test all features on the main Pro Dashboard page to ensure nothing has changed and no new errors show up. Thoroughly test the sites table, bulk management interface, and settings for each product (e.g., Backup, Scan, Monitor).
